### PR TITLE
Update calculation of secondary structure fraction

### PR DESCRIPTION
**Pull request by arpansahoo**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit``
locally, and understand that continuous integration checks will be used to
confirm the Biopython unit tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

I believe there is a mistake in secondary_structure_fraction() method in ProtParam. "V, I, Y, F, W, L" and "E, M, A, L" are listed as the amino acids that tend to be in helix and sheet respectively, but it is actually the opposite.

I have fixed this mistake, and I also updated the amino acid list for each secondary structure based on information found in the scientific articles below:

For beta turn: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2142776/pdf/7756980.pdf (Table 2)
- Added D (aspartic acid) to the list

For alpha helix: https://www.nature.com/articles/srep38341
- Added K (lysine)

For beta sheet: https://www.nature.com/articles/362267a0 (Table 1)
- Added T (threonine)

### CODE REVIEW

**Comment by arpansahoo** [comment]
@peterjc Could you take a look?

**Comment by peterjc** [comment]
You appear to be right, meaning that for almost 20 years (since Biopython 1.24) rather than returning an estimate of (Helix, Turn, Sheet) it has actually been returning (Sheet, Turn, Helix):

https://github.com/biopython/biopython/blob/biopython-124/Bio/SeqUtils/ProtParam.py#L220

That's possible, but strange enough that I'd like a second opinion on this - ideally from someone currently doing structural biology. Maybe @jgreener64 or @JoaoRodrigues?

It might be less disruptive to change the documentation to say it returns (Sheet, Turn, Helix) but historically wrongly claimed to do the opposite order?

You would also need to update the tests too.

Either way, I think this needs a line or two in the NEWS file.

**Comment by jgreener64** [comment]
From a quick look the PR seems correct. I don't work with these sequence tools much though.

**Comment by jgreener64** [comment]
> I didn't want to change the return to `sheet, turn, helix` in case any software out there does something like `helix, turn, sheet = X.secondary_structure_fraction()`

I would agree with this approach.

File: NEWS.rst [hunk 1]
```diff
@@ -18,12 +18,17 @@ iterators of the alignments to reset as well, which is bug prone. Instead,
 calling ``iter`` on a ``PairwiseAlignments`` object will now return itself. The
 iterator can be reset by calling the ``rewind`` method.
 
+Calling `secondary_structure_fraction` on a ``ProtParam.ProteinAnalysis``
+object historically returned (sheet, turn, helix) while claiming to return
+(helix, turn, sheet). This was fixed to correctly return (helix, turn, sheet).
+
```
Comment (peterjc) [inline NEWS.rst:24]: Add something like "... and the amino acids considered was revised as per recent literature."?

Comment (arpansahoo) [reply NEWS.rst:25]: Additionally, the amino acids considered were revised as per recent literature.

File: Bio/SeqUtils/ProtParam.py [hunk 2]
```diff
@@ -315,17 +319,20 @@ def secondary_structure_fraction(self):
         Returns a list of the fraction of amino acids which tend
         to be in Helix, Turn or Sheet.
```
[the remark this answers is not in the document]
Comment (arpansahoo) [reply Bio/SeqUtils/ProtParam.py:320]: I think that's a good idea. Would it be best to just add "according to Haimov and Srebnik, 2016; Hutchinson and Thornton, 1994; and Kim and Berg, 1993, respectively" at the end of "Returns a list of the fraction of amino acids which tend to be in Helix, Turn or Sheet"?

File: Bio/SeqUtils/ProtParam.py [hunk 3]
```diff
@@ -313,19 +317,23 @@ def secondary_structure_fraction(self):
         """Calculate fraction of helix, turn and sheet.
 
         Returns a list of the fraction of amino acids which tend
-        to be in Helix, Turn or Sheet.
+        to be in Helix, Turn or Sheet, according to Haimov and Srebnik, 2016;
+        Hutchinson and Thornton, 1994; and Kim and Berg, 1993, respectively.
```
Comment (arpansahoo) [inline Bio/SeqUtils/ProtParam.py:321]: Does this look good? @peterjc

**Comment by peterjc** [comment]
Nice. Are you happy for me to squash-and-merge this into a single commit (simpler history)?

**Comment by arpansahoo** [comment]
> Nice. Are you happy for me to squash-and-merge this into a single commit (simpler history)?

Sounds good to me

**Comment by peterjc** [comment]
Merged, thank you!

**Pull request merged**